server/src/main/java/org/elasticsearch/gateway/ClusterStateUpdaters.java

```diff
@@ -135,9 +135,11 @@ static ClusterState mixCurrentStateAndRecoveredState(final ClusterState currentS
         metadataBuilder.generateClusterUuidIfNeeded();
 
         for (final ProjectMetadata projectMetadata : recoveredState.metadata().projects().values()) {
+            final var projectBuilder = metadataBuilder.createNewProjectBuilder(projectMetadata.id());
             for (final IndexMetadata indexMetadata : projectMetadata) {
-                metadataBuilder.getProject(projectMetadata.id()).put(indexMetadata, false);
+                projectBuilder.put(indexMetadata, false);
```

Comment:
The change will create the Project builder if such project does not exist in the metadata already while the old node will throw NPE? I think we should avoid behaviour change?

Reply:
Good catch, thanks!